controllers/cluster_controller.go

```diff
@@ -287,3 +296,57 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileClusterStatus(ctx context.Context, cluster *clusterv1.Cluster) error {
+	if cluster.Status.ControlPlaneInitialized {
+		return nil
+	}
+
+	machines := &clusterv1.MachineList{}
+	err := r.Client.List(ctx, machines, client.InNamespace(cluster.Namespace))
```

Comment:
We should be able to also add a filter using the cluster-name

Reply:
s/should/must/

```go
client.MatchingLabels{clusterv1.MachineClusterLabelName: cluster.Name}
```